src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -235,6 +250,58 @@ const GenericError = () => {
   );
 };
 
+export const useOnRecurringPaymentReceived = ({
+  recurringPaymentId,
+  callback,
+}: {
+  recurringPaymentId: number | undefined;
+  callback: () => void;
+}) => {
+  const {userId} = useAuthContext();
+
+  const mapRecurringPayment = (
+    recurringPayment: FirebaseFirestoreTypes.DocumentData,
+  ): RecurringPayment => {
+    return {
+      id: recurringPayment.id,
+      expires_at: recurringPayment.expires_at,
+      masked_pan: recurringPayment.masked_pan,
+      payment_type: recurringPayment.payment_type,
+    };
+  };
+
+  useEffect(() => {
+    const recurringPaymentsUnsub = firestore()
```

Comment:
Have you checked how often this effect is run? I suspect it can be more than we want since the `callback` function isn't a stable reference.

Reply:
The only reason to have it here a useEffect is to the unsubscribe here the firebase listener, because if it is sent in a return, any better idea to do not use useeffect specially because of the callback function?

Reply:
We need the `useEffect`. Rather it is the `callback` function that needs to be a stable reference somehow. Maybe either through `useCallback`, or use the `useRef` pattern we have in utils like `useInterval` and `useDoOnceWhen`.